webapp/src/hook/hook.js

```diff
@@ -37,26 +41,55 @@ export default class Hooks {
             if (commands.length > 2) {
                 return Promise.resolve({error: {message: 'Too many arguments'}});
             }
-            privateKey = Buffer.from(commands[1], 'base64');
-            publicKey = getPublicKeyFromPrivateKey(privateKey);
-            if (!publicKey) {
+
+            key = keyFromString(commands[1]);
+            if (!key) {
                 return Promise.resolve({error: {message: 'Invalid private key'}});
             }
 
-            pubKeyString = publicKey.toString('base64');
-            storePrivateKey(privateKey);
-            return Promise.resolve({message: '/anonymous keypair --overwrite ' + pubKeyString, args});
+            storePrivateKey(key);
+            return Promise.resolve({message: '/anonymous keypair --overwrite ' + publicKeyToString(key), args});
 
         case '--export':
-            privateKey = getPrivateKey();
-            this.store.dispatch(sendEphemeralPost('Your private key:\n' + privateKey.toString('base64'), args.channel_id));
+            key = loadKey();
+            this.store.dispatch(sendEphemeralPost('Your private key:\n' + privateKeyToString(key), args.channel_id));
             return Promise.resolve({});
         default:
             break;
         }
         return Promise.resolve({message: '/anonymous keypair' + commands[0], args});
     };
 
+    handlePost = async (commands, args) => {
+        const users = await Client4.getProfilesInChannel(args.channel_id);
+
+        const publicKeys = await Promise.all(users.map((user) => {
+            return Client.retrievePublicKey(user.id).then((data) => {
+                return Buffer.from(data.public_key, 'base64').toString();
+            });
+        }));
+
+        const encrypted = await Promise.all(publicKeys.map((keyString) => {
+            const key = keyFromString(keyString);
+            const encrypter = newFromPublicKey(key);
+            const message = encrypter.encrypt(commands[0]).toString('base64');
```

Comment:
Here we only encrypt data before first space. If the message is `one two` we will only send `one`